### PR TITLE
Added U512T

### PR DESCRIPTION
**Pull request by Sniezka1927**

### CODE REVIEW

File: test/clamm.test.ts [hunk 1]
```diff
@@ -12,6 +12,27 @@ describe('math tests', () => {
...
+    }
+    {
+      const a = 115792089237316195423570985008687907853269984665640564039457584007913129639935n
+      const b = 999n
+      const result = (await clamm.contractInstance.methods.bigAdd({ args: { a, b } })).returns
+      expect(result).toStrictEqual({ higher: b, lower: a })
+    }
+    {
+      const a = 777n
+      const b = 115792089237316195423570985008687907853269984665640564039457584007913129639935n
```
Comment (wojciech-cichocki) [inline test/clamm.test.ts:31]: maybe add also one edge case: both a and b are `u256Max!()`

**Review by wojciech-cichocki — CHANGES_REQUESTED** [review]

**Review by wojciech-cichocki — APPROVED** [review]
